Modules/_testcapi/codec.c

```diff
@@ -1,17 +1,244 @@
 #include "parts.h"
-#include "util.h"
 
+// === Codecs registration and un-registration ================================
+
+static PyObject *
+codec_register(PyObject *Py_UNUSED(module), PyObject *search_function)
+{
+    if (PyCodec_Register(search_function) < 0) {
+        return NULL;
+    }
+    Py_RETURN_NONE;
+}
+
+static PyObject *
+codec_unregister(PyObject *Py_UNUSED(module), PyObject *search_function)
+{
+    if (PyCodec_Unregister(search_function) < 0) {
+        return NULL;
+    }
+    Py_RETURN_NONE;
+}
+
+static PyObject *
+codec_known_encoding(PyObject *Py_UNUSED(module), PyObject *args)
+{
+    const char *encoding;   // should not be NULL
+    if (!PyArg_ParseTuple(args, "z", &encoding)) {
```

Comment:
Use "s" if encoding must not be NULL, no?

Reply:
I always mix them because of their meaning in `PyBuild_Value`... Thanks!

Reply:
AH! I remember why I used "z" and not "s"! The PyCodec_Decode() API crashes if you specify a NULL pointer. So I wanted to emulate that behaviour without `PyArg_ParseTuple` being the one that is telling me that the input is invalid.

Was it ok to do it like this or not?